### PR TITLE
Fixes problems in SQLite Adapter in foreign key creation

### PR DESCRIPTION
**Pull request by charlienux**
This PR fixes two problems #1840 and #1841.

### CODE REVIEW

**Comment by dereuromark** [comment]
It looks like there is some error left here in the tests.

File: src/Phinx/Db/Adapter/SQLiteAdapter.php [hunk 1]
```diff
@@ -47,7 +47,6 @@ class SQLiteAdapter extends PdoAdapter
         self::PHINX_TYPE_CHAR => 'char',
         self::PHINX_TYPE_DATE => 'date_text',
         self::PHINX_TYPE_DATETIME => 'datetime_text',
-        self::PHINX_TYPE_DECIMAL => 'decimal',
```
Comment (dereuromark) [inline src/Phinx/Db/Adapter/SQLiteAdapter.php:50]: Are you sure we need to remove this type here? It can have other side effects and regressions.

Comment (charlienux) [reply src/Phinx/Db/Adapter/SQLiteAdapter.php]: Indeed that line of code should not be removed.

I reversed the change.

**Comment by charlienux** [comment]
> It looks like there is some error left here in the tests.

Errors in the tests are due to the line removed by mistake.

**Pull request closed without merging**